template/web/src/hooks/useCollectionMetadata.ts

```diff
@@ -1,115 +1,145 @@
-import { useCallback, useEffect, useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
 import { Abi, Address } from 'abitype';
-import { useReadContract } from 'wagmi';
+import { Chain } from 'viem/chains';
+import { useAccount, usePublicClient } from 'wagmi';
 import { ipfsToHTTP } from '@/utils/ipfs';
 
+// Minimal ABI with contractURI view function
+const AbiWithContractURI = [
```

Comment:
can we put the ABI's in files similar to this: https://github.com/coinbase/build-onchain-apps/blob/4d9b38cfefb1dceff36aee0a229168a9a01971b4/template/web/app/mint/_contracts/Custom1155ABI.ts#L2?

Reply:
hmm I spent a bit of time on moving these abis as well as validation utils as a separate utils and it doesn't feel right to me I'll be honest. 

These are not really abis, I should probably rename them to `AbiFunctionContractURI` and `AbiFunctionURI`. As these are more like fragments of the ABIs, and shouldn't be used as abis by mistake.
I kind of like all of this code to be self-contained within the same, especially since it has very specific and somewhat hacky way to check which function to use on the abi (contractURI vs uri). 

I just pushed a change where I move the action away from the hook at least. Let me know if this is better